### PR TITLE
SNOW-749232: Upgrade to aws-sdk v3

### PR DESCRIPTION
**Pull request by sfc-gh-dprzybysz**
### Description
aws-sdk v2 causes warning message:

```
(node:39752) NOTE: We are formalizing our plans to enter AWS SDK for JavaScript (v2) into maintenance mode in 2023.

Please migrate your code to use AWS SDK for JavaScript (v3).
For more information, check the migration guide at https://a.co/7PzMCcy
(Use `node --trace-warnings ...` to show where the warning was created)
```

and we should switch to AWS SDK in version 3.

**AWS SDK v3 needs Node version `>=14`**

### Checklist
- [x] Format code according to the existing code style (run `npm run lint:check -- CHANGED_FILES` and fix problems in changed code)
- [x] Create tests which fail without the change (if possible)
- [x] Make all tests (unit and integration) pass (`npm run test:unit` and `npm run test:integration`)
- [x] Extend the README / documentation and ensure is properly displayed (if necessary)
- [x] Provide JIRA issue id (if possible) or GitHub issue id in commit message

### CODE REVIEW

**Comment by sahilkul95** [comment]
@sfc-gh-pfus @sfc-gh-igarish
Can we please prioritize this? We are blocked for only one upgrade.

**Comment by sfc-gh-dprzybysz** [comment]
> @sfc-gh-pfus @sfc-gh-igarish Can we please prioritize this? We are blocked for only one upgrade.

We are planning to include it in the next month release

**Review by sfc-gh-pfus — APPROVED** [review]

File: lib/file_transfer_agent/s3_util.js [hunk 1]
```diff
@@ -22,157 +21,134 @@ const resultStatus = require('./file_util').resultStatus;
...
-  * @returns {Object}
-  */
-  this.getFileHeader = async function (meta, filename)
-  {
-    var stageInfo = meta['stageInfo'];
-    var client = this.createClient(stageInfo);
-    var s3location = this.extractBucketNameAndPath(stageInfo['location']);
-
-    var params = {
+   * Create file header based on file being uploaded or not.
```
Comment (sfc-gh-pmotacki) [inline lib/file_transfer_agent/s3_util.js:108]: It's a minor but for me we rather extract/get this data not create. WDYT?

**Review by sfc-gh-pmotacki — APPROVED** [review]

**Comment by sahilkul95** [comment]
Hi team, are we planning to release this in this month? Any updates?

**Pull request merged**